File: to_do_list.go
package main

import (
	"fmt"
	"os"
	"strconv"
	"time"

	"gorm.io/driver/sqlite"
	"gorm.io/gorm"
)

type Task struct {
	ID       uint `gorm:"primaryKey"`
	Task     string
	Deadline time.Time
}

func main() {
	db, err := gorm.Open(sqlite.Open("todo.db"), &gorm.Config{})
	if err != nil {
		panic("failed to connect database")
	}
	db.AutoMigrate(&Task{})

	for {
		var choice string
		fmt.Println("1) Today's tasks")
		fmt.Println("2) Week's tasks")
		fmt.Println("3) All tasks")
		fmt.Println("4) Missed tasks")
		fmt.Println("5) Add task")
		fmt.Println("6) Delete task")
		fmt.Println("0) Exit")
		fmt.Scan(&choice)

		switch choice {
		case "0":
			os.Exit(0)
		case "1":
			today := time.Now()
			rows := []Task{}
			db.Where("deadline BETWEEN ? AND ?", today, today.Add(24*time.Hour)).Find(&rows)
			printTasks(rows)
		case "2":
			for i := 0; i < 7; i++ {
				day := time.Now().AddDate(0, 0, i)
				rows := []Task{}
				db.Where("deadline BETWEEN ? AND ?", day, day.Add(24*time.Hour)).Find(&rows)
				fmt.Printf("%s %s %d:\n", day.Weekday(), day.Month(), day.Day())
				printTasks(rows)
				fmt.Println()
			}
		case "3":
			rows := []Task{}
			db.Order("deadline").Find(&rows)
			printTasksWithDate(rows)
		case "4":
			today := time.Now()
			rows := []Task{}
			db.Where("deadline < ?", today).Find(&rows)
			printTasksWithDate(rows)
		case "5":
			addTask(db)
		case "6":
			deleteTask(db)
		default:
			fmt.Println("Invalid choice. Please try again.")
		}
	}
}

func printTasks(tasks []Task) {
	if len(tasks) == 0 {
		fmt.Println("Nothing to do!")
		return
	}
	for i, task := range tasks {
		fmt.Printf("%d. %s\n", i+1, task.Task)
	}
}

func printTasksWithDate(tasks []Task) {
	if len(tasks) == 0 {
		fmt.Println("Nothing to do!")
		return
	}
	for i, task := range tasks {
		fmt.Printf("%d. %s. %s\n", i+1, task.Task, task.Deadline.Format("02 Jan"))
	}
}

func addTask(db *gorm.DB) {
	var taskString, deadlineString string
	fmt.Println("Enter task:")
	fmt.Scan(&taskString)
	fmt.Println("Enter deadline (YYYY-MM-DD):")
	fmt.Scan(&deadlineString)

	deadline, err := time.Parse("2006-01-02", deadlineString)
	if err != nil {
		fmt.Println("Invalid date format. Please use YYYY-MM-DD format.")
		return
	}

	task := Task{
		Task:     taskString,
		Deadline: deadline,
	}
	db.Create(&task)
	fmt.Println("The task has been added!")
}

func deleteTask(db *gorm.DB) {
	var idString string
	fmt.Println("Choose the number of the task you want to delete:")
	fmt.Scan(&idString)
	id, err := strconv.Atoi(idString)
	if err != nil {
		fmt.Println("Invalid input. Please enter a valid number.")
		return
	}

	var task Task
	db.First(&task, id)
	db.Delete(&task)
	fmt.Println("The task has been deleted!")
}

File: to_do_list.py
from sqlalchemy.ext.declarative import declarative_base
from sqlalchemy import create_engine, Column, Integer, String, Date
from datetime import datetime, timedelta
from sqlalchemy.orm import sessionmaker

engine = create_engine('sqlite:///todo.db?check_same_thread=False')
Base = declarative_base()


class Table(Base):
    __tablename__ = 'task'
    id = Column(Integer, primary_key=True)
    task = Column(String, default='default_value')
    deadline = Column(Date, default=datetime.today())

    def __repr__(self):
        return self.task


while True:
    Base.metadata.create_all(engine)
    Session = sessionmaker(bind=engine)
    session = Session()
    rows = session.query(Table).all()
    today = datetime.today()
    print("1) Today's tasks")
    print("2) Week's tasks")
    print('3) All tasks')
    print('4) Missed tasks')
    print('5) Add task')
    print('6) Delete task')
    print('0) Exit')
    choice = input()
    if choice == '0':
        break
    elif choice == '1':
        rows = session.query(Table).filter(Table.deadline == today.date()).all()
        print(f"Today {today.day} {today.strftime('%b')}:")
        if not rows:
            print('Nothing to do!')
        else:
            counter = 1
            for row in rows:
                print(f'{counter}. {row.task}')
                counter += 1
        print()
    elif choice == '2':
        for i in range(7):
            next_day = datetime.today() + timedelta(days=i)
            day_rows = session.query(Table).filter(Table.deadline == next_day.date()).all()
            print(f"{next_day.strftime('%A')} {next_day.day} {next_day.strftime('%b')}:")
            if not day_rows:
                print('Nothing to do!')
            else:
                counter = 1
                for row in day_rows:
                    print(f'{counter}. {row.task}')
                    counter += 1
            print()
    elif choice == '3':
        rows = session.query(Table).order_by(Table.deadline).all()
        print('All tasks:')
        if not rows:
            print('Nothing to do!')
        else:
            counter = 1
            for row in rows:
                print(f"{counter}. {row.task}. {row.deadline.day} {row.deadline.strftime('%b')}")
                counter += 1
        print()
    elif choice == '4':
        rows = session.query(Table).filter(Table.deadline < today.date()).all()
        print('Missed tasks:')
        if not rows:
            print('Nothing is missed!')
        else:
            counter = 1
            for row in rows:
                print(f"{counter}. {row.task}. {row.deadline.day} {row.deadline.strftime('%b')}")
                counter += 1
        print()
    elif choice == '5':
        try:
            print('Enter task')
            new_task = input()
            print('Enter deadline')
            new_deadline = input()
            dl = datetime.strptime(new_deadline, '%Y-%m-%d')
            new_row = Table(task=new_task, deadline=dl)
            session.add(new_row)
            session.commit()
            print('The task has been added!')
        except ValueError:
            print('Deadline should match the format YYYY-MM-DD')
        print()
    elif choice == '6':
        rows = session.query(Table).order_by(Table.deadline).all()
        print('Choose the number of the task you want to delete:')
        if not rows:
            print('Nothing to delete!')
        else:
            counter = 1
            for row in rows:
                print(f"{counter}. {row.task}. {row.deadline.day} {row.deadline.strftime('%b')}")
                counter += 1
            del_num = int(input())
            session.delete(rows[del_num-1])
            session.commit()
        print()
